resources/views/standings/upload.blade.php

```diff
@@ -1,241 +1,3533 @@
+            if(selectedValue === null || selectedValue === '') {
+                $(parentNode).removeClass('bg-yellow-600');
+                $(parentNode).addClass('bg-red-600');
+                $(alertNode).slideDown(500);
+            }
+            else {
+                $(parentNode).removeClass('bg-red-600');
+                $(alertNode).slideUp(500);
+            }
+        }
+    }
+
+    function repopulateConstructorResultsDropdowns(json, supportingVariables) {
+        const e = new Event("change");
+
+        for(let x = 0; x < json.results.length; x++) {
+            let constructorSelect = "<option hidden selected value> -- Missing ID -- </option>";
+            let constructorDispVal;
+            let constructorCol = document.getElementById(`constructorSelect${x}`);
+
+            for(let y = 0; y < supportingVariables.availableConstructors.length; y++) {
+                constructorDispVal = `${supportingVariables.availableConstructors[y].id} - ${supportingVariables.availableConstructors[y].name}`;
+
+                if(supportingVariables.availableConstructors[y].id !== json.results[x].constructor_id) {
+                    constructorSelect += `<option value=${supportingVariables.availableConstructors[y].id}>${constructorDispVal}</option>`;
+                }
+                else {
+                    constructorSelect += `<option selected value=${supportingVariables.availableConstructors[y].id}>${constructorDispVal}</option>`;
+                }
+                constructorCol.innerHTML = constructorSelect;
+            }
+            constructorCol.dispatchEvent(e);
+        }
+    }
+
+    function checkAllTableValuesForErrors(json, regexValidationStrings, jsonResultsDetailsStore, supportingVariables) {
+        let postStatus = {
+            track: 1,
+            results: 1
+        }
+
+        checkEachCellValueForError(json, postStatus);
+
+        if(
+            isValidTimeFormat(json, regexValidationStrings, jsonResultsDetailsStore) && 
+            (postStatus.track === 1) && 
+            (postStatus.results === 1)
+        ) {
+            $('#errorSubmitJSONAlert').slideUp(500);
+            $('#submitJSON').removeClass('disableActionBtns');
+            $('#reviewJSON').removeClass('disableActionBtns');
+        }
+        else {
+            $('#submitJSON').addClass('disableActionBtns');
+            $('#reviewJSON').addClass('disableActionBtns');
+            $('#errorSubmitJSONAlert').html('<p>Please clear all the <strong>ERRORS</strong></p>');
+            $('#errorSubmitJSONAlert').slideDown(500);
+        }
+    }
+
+    function checkEachCellValueForError(json, postStatus) {
+        let trackOuterDivs = ['#trackBodySeason', '#trackBodyRound', '#trackBodyCircuit', '#trackBodyPoints'];
+
+        for(let i = 0; i < trackOuterDivs.length; i++) {
+            postStatus.track -= checkForError(trackOuterDivs[i]);
+        }
+
+        for(let i = 0; i < json.results.length; i++){
```

Comment:
instead of loops inside loops probably look into Array.map(), Array.filter() or Array.reduce() they might be helpful in this case

Reply:
Noted.
Array prototype functions are faster and more readable.